cmd/soroban-cli/src/commands/contract/build.rs

```diff
@@ -199,27 +205,63 @@ impl Cmd {
                     return Err(Error::Exit(status));
                 }
 
-                let file = format!("{}.wasm", p.name.replace('-', "_"));
-                let target_file_path = Path::new(target_dir)
-                    .join(&wasm_target)
-                    .join(&self.profile)
-                    .join(&file);
+                self.build_package(&print, &p, target_dir, &wasm_target)?;
+            }
+        }
 
-                self.handle_contract_metadata_args(&target_file_path)?;
+        Ok(())
+    }
 
-                let final_path = if let Some(out_dir) = &self.out_dir {
-                    fs::create_dir_all(out_dir).map_err(Error::CreatingOutDir)?;
-                    let out_file_path = Path::new(out_dir).join(&file);
-                    fs::copy(target_file_path, &out_file_path).map_err(Error::CopyingWasmFile)?;
-                    out_file_path
-                } else {
-                    target_file_path
-                };
+    fn build_package(
+        &self,
+        print: &Print,
+        package: &Package,
+        target_dir: &Utf8PathBuf,
+        wasm_target: &str,
+    ) -> Result<(), Error> {
+        let file = format!("{}.wasm", package.name.replace('-', "_"));
+        let target_file_path = Path::new(target_dir)
+            .join(wasm_target)
+            .join(&self.profile)
+            .join(&file);
+
+        self.handle_contract_metadata_args(&target_file_path)?;
+
+        let final_path = if let Some(out_dir) = &self.out_dir {
+            fs::create_dir_all(out_dir).map_err(Error::CreatingOutDir)?;
+            let out_file_path = Path::new(out_dir).join(&file);
+            fs::copy(&target_file_path, &out_file_path).map_err(Error::CopyingWasmFile)?;
+            out_file_path
+        } else {
+            target_file_path.clone()
+        };
 
-                Self::print_build_summary(&print, &final_path)?;
+        #[cfg(feature = "additional-libs")]
+        let mut optimized_path = None;
+        #[cfg(not(feature = "additional-libs"))]
+        let optimized_path = None;
+
+        if self.optimize {
```

Comment:
Instead of creating multiple variables, one which is mutable, you can have the value built in the block returned into the value.
```suggestion
        let optimized_path = if self.optimize {
```